control-plane/agents/src/bin/core/volume/specs.rs

```diff
@@ -173,6 +181,54 @@ pub(crate) async fn nexus_attach_candidates(
     Ok(candidates)
 }
 
+/// Return a list of appropriate requests which can be used to create a replica on a pool.
+/// This can be used when the volume's current replica count is smaller than the desired volume's
+/// replica count.
+pub(crate) async fn volume_replica_candidates_new(
+    registry: &Registry,
+    volume_spec: &VolumeSpec,
+) -> Result<Vec<Vec<CreateReplicaNew>>, SvcError> {
```

Comment:
This doesn't seem right, again it's adding label per replica, rather than per vector of create_requests
Also use the `newtype` pattern rather than repeating all fields in the new definitions.
I think it should be more like this:
```rust
Result<Vec<CreateReplicas>, SvcError>....

struct CreateReplicas {
   replicas: Vec<CreateReplica>,
   affinity_labels: HashMap<String, String>,
}
```